test/EFCore.PG.Tests/Migrations/MigrationSqlGeneratorTestBase.cs

```diff
@@ -0,0 +1,172 @@
+                    MaxLength = 32,
+                    IsNullable = true
+                });
+
+        [ConditionalFact]
+        public virtual void AddColumnOperation_with_maxLength_no_model()
+            => Generate(
+                new AddColumnOperation
+                {
+                    Table = "Person",
+                    Name = "Name",
+                    ClrType = typeof(string),
+                    MaxLength = 30,
+                    IsNullable = true
+                });
+
+        [ConditionalFact]
+        public virtual void AddForeignKeyOperation_without_principal_columns()
+            => Generate(
+                new AddForeignKeyOperation
+                {
+                    Table = "People",
+                    Columns = new[] { "SpouseId" },
+                    PrincipalTable = "People"
+                });
+
+        [ConditionalFact]
+        public virtual void AlterColumnOperation_without_column_type()
+            => Generate(
+                new AlterColumnOperation
+                {
+                    Table = "People",
+                    Name = "LuckyNumber",
+                    ClrType = typeof(int)
+                });
+
+        [ConditionalFact]
+        public virtual void RenameTableOperation_legacy()
+            => Generate(
+                new RenameTableOperation
+                {
+                    Name = "People",
+                    Schema = "dbo",
+                    NewName = "Person"
+                });
+
+        [ConditionalFact]
+        public virtual void RenameTableOperation()
+            => Generate(
+                modelBuilder => modelBuilder.HasAnnotation(CoreAnnotationNames.ProductVersion, "2.1.0"),
+                new RenameTableOperation
+                {
+                    Name = "People",
+                    Schema = "dbo",
+                    NewName = "Person",
+                    NewSchema = "dbo"
+                });
+
+        [ConditionalFact]
+        public virtual void SqlOperation()
+            => Generate(
+                new SqlOperation { Sql = "-- I <3 DDL" });
+
+        protected TestHelpers TestHelpers { get; }
+
+        protected MigrationSqlGeneratorTestBase(TestHelpers testHelpers)
+        {
+            TestHelpers = testHelpers;
+        }
+
+        protected virtual void Generate(params MigrationOperation[] operation)
+            => Generate(_ => { }, operation);
+
+        protected virtual void Generate(Action<ModelBuilder> buildAction, params MigrationOperation[] operation)
+        {
+            var modelBuilder = TestHelpers.CreateConventionBuilder();
+            modelBuilder.Model.RemoveAnnotation(CoreAnnotationNames.ProductVersion);
+            buildAction(modelBuilder);
+
+            var batch = TestHelpers.CreateContextServices().GetRequiredService<IMigrationsSqlGenerator>()
+                .Generate(operation, modelBuilder.Model);
+
+            Sql = string.Join(
+                "GO" + EOL + EOL,
```

Comment:
Is see an SQL Server directive!

Reply:
This is just a "delimiter" introduced in the tests to indicate a batch boundary (I've added a comment) - yes it's "inspired" by SQL Server :) Don't forget that these are just unit tests on the produced migrations DDL - nothing is executed on the database. Also note that this base class is copied as-is from EF Core, since there's no nuget package for the non-functional tests.